tests/test_improved_anchor_matching.py

```
@@ -0,0 +1,224 @@
+                
+                # Verify it's near the end, not in the middle
+                total_lines = len(lines)
+                if i >= total_lines - 5:  # Within last 5 lines
+                    print("✅ SUCCESS: Method inserted near class end!")
+                    return True
+                else:
+                    print("❌ FAIL: Method inserted too early in file")
+                    return False
+                    
+        print("❌ FAIL: NewMethod not found in result")
+        return False
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)